File: codegen-tool/src/main/java/com/dqjq/codegen/Generator/RepositoryGenerator.java
package com.dqjq.codegen.Generator;

import com.dqjq.codegen.CodegenConfig;
import lombok.Getter;

import java.io.IOException;
import java.nio.file.Path;

/**
 * Repository 接口生成器
 * <p>
 * 负责生成 JPA Repository 接口。
 * </p>
 */
@Getter
public class RepositoryGenerator extends BaseGenerator {

    private Path repoPath;

    public RepositoryGenerator(CodegenConfig cfg) {
        super(cfg);
    }

    @Override
    public void generate() throws IOException {
        Path baseOut = getBaseOutputDir();
        String basePackagePath = cfg.packageBase.replace('.', '/');
        String modulePath = baseOut + "/" + basePackagePath + "/" + cfg.module;
        Path repoDir = projectRoot.resolve(modulePath + "/repository");
        
        ensureDirectory(repoDir);
        
        this.repoPath = repoDir.resolve(cfg.entityName + "Repo.java");

        String template = readTemplate("repository.tpl");
        if (template.isEmpty()) {
            throw new IOException("无法加载 Repository 模板: repository.tpl");
        }

        String repoPkg = cfg.packageBase + "." + cfg.module + ".repository";
        String entityPkg = cfg.packageBase + "." + cfg.module + ".entity";

        String content = template
                .replace("{{package}}", repoPkg)
                .replace("{{entityPackage}}", entityPkg)
                .replace("{{entityName}}", cfg.entityName)
                .replace("{{idType}}", cfg.idType);

        writeFile(repoPath, content);
    }

}

File: codegen/codegen-tool-py/codegen/generators/repository_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Repository生成器
生成Java Repository接口
"""

from pathlib import Path
from codegen.generators.base_generator import BaseGenerator

class RepositoryGenerator(BaseGenerator):
    """Repository生成器"""
    
    def generate(self):
        """
        生成Repository接口
        """
        # 准备上下文数据
        context = {
            'package': self.get_full_package('repository'),
            'entity_name': self.code_generator.entity_name,
            'id_type': self.code_generator.id_type,
            'entity_package': self.get_full_package('entity'),
            'repository_base_class': self.config_dict.get('repositoryBaseClass', ''),
            'has_base_class': bool(self.config_dict.get('repositoryBaseClass', ''))
        }
        
        # 渲染模板
        content = self.render_template('java/repository.j2', context)
        
        # 生成文件路径
        output_dir = self.code_generator.get_output_path('repository')
        self.output_path = output_dir / f"{self.code_generator.entity_name}Repository.java"
        
        # 写入文件
        self.write_file(self.output_path, content)
